st2api/st2api/controllers/exp/inquiries.py

```diff
@@ -0,0 +1,295 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_config import cfg
+
+import copy
+import jsonschema
+import json
+
+from six.moves import http_client
+from st2common.models.db.auth import UserDB
+from st2api.controllers.resource import ResourceController
+from st2api.controllers.v1.executionviews import SUPPORTED_FILTERS
+from st2common import log as logging
+from st2common.util import action_db as action_utils
+from st2common.util import system_info
+from st2common.services import executions
+from st2common.constants import action as action_constants
+from st2common.rbac.types import PermissionType
+from st2common.rbac import utils as rbac_utils
+from st2common.router import abort
+from st2common.models.api.execution import ActionExecutionAPI
+from st2common.persistence.execution import ActionExecution
+from st2common.services import action as action_service
+from st2common.util.action_db import (get_action_by_ref, get_runnertype_by_name)
+
+
+from st2actions.container.base import get_runner_container
+
+__all__ = [
+    'InquiriesController'
+]
+
+LOG = logging.getLogger(__name__)
+
+
+class InquiriesController(ResourceController):
+    """API controller for Inquiries
+    """
+
+    supported_filters = copy.deepcopy(SUPPORTED_FILTERS)
+
+    # No data model currently exists for Inquiries, so we're "borrowing" ActionExecutions
+    model = ActionExecutionAPI
+    access = ActionExecution
+
+    def get_all(self, requester_user=None, limit=None, **raw_filters):
+        """Retrieve multiple Inquiries
+
+           TODO(mierdin): This function doesn't currently make RBAC assertions for two reasons:
+           - The equivalent for executions (Resource._get_all) doesn't seem to do this either.
+           - I am not sure the best approach, since I would have to make an assertion
+             inside the below loop?
+
+            Handles requests:
+                GET /inquiries/
+        """
+
+        raw_inquiries = super(InquiriesController, self)._get_all(
+            limit=limit,
+            raw_filters={'status': 'pending', 'runner': 'inquirer'}
```

Comment:
Should probably reference a constant for "pending" here.

Reply:
Changed in 66f3fb77a